File: DemoServer/AdditionalLanguages/go/Revisions/getRevisions/GetRevisions.go
package main

//region Usings
import "github.com/ravendb/ravendb-go-client"
//endregion

var globalDocumentStore *ravendb.DocumentStore

func main() {
    createDocumentStore()
    createDatabase()
    getRevisions()
    globalDocumentStore.Close()
}

func createDocumentStore() (*ravendb.DocumentStore, error) {
    if globalDocumentStore != nil {
        return globalDocumentStore, nil
    }
    urls := []string{"http://localhost:8080"}
    store := ravendb.NewDocumentStore(urls, "testGO")
    err := store.Initialize()
    if err != nil {
        return nil, err
    }
    globalDocumentStore = store
    return globalDocumentStore, nil
}

func createDatabase() {
    databaseRecord := ravendb.NewDatabaseRecord()
    databaseRecord.DatabaseName = "testGO"
    createDatabaseOperation := ravendb.NewCreateDatabaseOperation(databaseRecord, 1)
    var err = globalDocumentStore.Maintenance().Server().Send(createDatabaseOperation)
    if err != nil {
        fmt.Printf("d.store.Maintenance().Server().Send(createDatabaseOperation) failed with %s\n", err)
    }
}

type Company struct {
    ID      string
    Name    string
    Phone   string
    Contact *Contact
}

type Contact struct {
    Name  string
    Title string
}

//region Demo
func getRevisions() error {

    //region Step_1
    myRevisionsConfiguration := &ravendb.RevisionsConfiguration {
        DefaultConfig: &ravendb.RevisionsCollectionConfiguration {
            Disabled: false,
        },
    }

    revisionsConfigurationOperation := ravendb.NewConfigureRevisionsOperation(myRevisionsConfiguration)
    err := globalDocumentStore.Maintenance().Send(revisionsConfigurationOperation)
    if err != nil {
        return nil
    }
    //endregion
    
    session, err := globalDocumentStore.OpenSession("")
    if err != nil {
        return err
    }
    defer session.Close()
    
    //region Step_2
    var company *Company
    err = session.Load(&company, "companies/7-A")
    if err != nil {
        return err
    }
    
    company.Name = "Name 1"
    err = session.SaveChanges()
    if err != nil {
        return err
    }
    
    company.Name = "Name 2"
    company.Phone = "052-1234-567"
    err = session.SaveChanges()
    if err != nil {
        return err
    }
    //endregion
    
    //region Step_3
    var revisions []*Company
    err = session.Advanced().Revisions().GetFor(&revisions, "companies/7-A")
    if err != nil {
        return err
    }
    //endregion
    
    return nil
}
//endregion

File: DemoServer/AdditionalLanguages/php/revisions/getRevisions/GetRevisions.php
<?php

namespace RavenDB\Demo\revisions\getRevisions;

//region Usings
use RavenDB\Documents\Operations\Revisions\ConfigureRevisionsOperation;
use RavenDB\Documents\Operations\Revisions\RevisionsCollectionConfiguration;
use RavenDB\Documents\Operations\Revisions\RevisionsConfiguration;
//endregion

use RavenDB\Demo\common\DocumentStoreHolder;
use RavenDB\Demo\common\models\Company;

class GetRevisions
{
    public function __invoke(): array
    {
        //region Demo
        //region Step_1
        $myRevisionsConfiguration = new RevisionsConfiguration();

        $defaultConfiguration = new RevisionsCollectionConfiguration();
        $defaultConfiguration->setDisabled(false);
        $myRevisionsConfiguration->setDefaultConfig($defaultConfiguration);

        $revisionsConfigurationOperation = new ConfigureRevisionsOperation($myRevisionsConfiguration);
        DocumentStoreHolder::getStore()->maintenance()->send($revisionsConfigurationOperation);
        //endregion

        $revisions = [];
        $session = DocumentStoreHolder::getStore()->openSession();
        try {
        
            //region Step_2
            $company = $session->load(Company::class, "companies/7-A");

            $company->setName("Name 1");
            $session->countersFor("companies/7-A")->increment("MyCounter", 100);
            $session->saveChanges();

            $company->setName("Name 2");
            $company->setPhone("052-1234-567");
            $session->saveChanges();
            //endregion

            //region Step_3
            $revisions = $session
                ->advanced()
                ->revisions()
                ->getFor(Company::class, "companies/7-A", 0, 25);
            //endregion
            
        } finally {
            $session->close();
        }
        //endregion

        return $revisions;
    }
}
